ERCS/erc-7825.md

```diff
@@ -0,0 +1,160 @@
+
+## Specification
+
+### Structs
+
+The State Channel Protocol defines the following key structs:
+
+#### `FixedPart`
+Defines immutable parameters of the state channel:
+
+```solidity
+struct FixedPart {
+    address[]   participants;        // List of participant addresses
+    uint48      channelNonce;        // Unique channel identifier
+    address     appDefinition;       // Address of the application-specific contract
+    uint48      challengeDuration;   // Challenge duration in seconds
+}
+```
+
+#### `VariablePart`
+Defines the dynamic parameters of a channel's state:
+
+```solidity
+struct VariablePart {
+    ExitFormat.SingleAssetExit[] outcome;   // Asset distribution upon channel finalization
+    bytes   appData;                        // Application-specific data
+    uint48  turnNum;                        // State version number
+    bool    isFinal;                        // Flag for instant finalization
+}
+```
+
+#### `SignedVariablePart`
+Represents a signed state update:
+
+```solidity
+struct SignedVariablePart {
+    VariablePart variablePart;
+    Signature[]  sigs;                      // Participant signatures
+}
+```
+
+#### `Outcome`
+Describes asset distribution after channel finalization:
+
+```solidity
+struct Outcome {
+    SingleAssetExit[] singleAssetExits;
+}
+
+struct SingleAssetExit {
+    address         asset;                         // Asset type (native or ERC20)
+    bytes           assetMetadata;                   // Metadata for exotic assets
+    Allocation[]    allocations;              // Distribution of funds
+}
+
+struct Allocation {
+    bytes32 destination;                   // Destination (channel ID or external address)
+    uint256 amount;                        // Allocation amount
+    uint8   allocationType;                  // Type (0 = simple, 1 = guarantee)
+    bytes   metadata;                        // Additional data for guarantees
+}
+```
+
+### Interfaces
+
+#### `IForceMoveApp`
+Defines the application-specific execution rules:
+
+```solidity
+interface IForceMoveApp {
+    function stateIsSupported(
+        FixedPart calldata fixedPart,
+        RecoveredVariablePart[] calldata proof,
+        RecoveredVariablePart calldata candidate
+    ) external view returns (bool, string memory);
+}
+```
+
+#### `IAdjudicator`
+Handles on-chain dispute resolution:
+
+```solidity
+interface IAdjudicator {
+    function challenge(
+        bytes32 channelId,
+        FixedPart calldata fixedPart,
+        SignedVariablePart[] calldata proof,
+        SignedVariablePart calldata candidate
+    ) external;
+
+    function finalize(bytes32 channelId) external;
+}
+```
+
+### Integration with ERC-4337
+
+#### Benefits
```

Comment:
Anything not describing requirements should be in the rationale or motivation section.